pre_cleanup.lua

```diff
@@ -0,0 +1,47 @@
+local os = require('os')
+
+local function clean_state(self)
+    box.space._space:pairs():map(function(tuple)
+        return tuple[3]
+    end):filter(function(name)
+        local first_char = string.sub(name, 1, 1)
+        return first_char ~= '_'
+    end):each(function(name)
+        box.space[name]:drop()
+    end)
+    local allowed_users = {
+        guest = true,
+        admin = true,
+    }
+
+    box.space._user:pairs():filter(function(tuple)
+        return tuple[4] == 'user'
+    end):map(function(tuple)
+        return tuple[3]
+    end):filter(function(name)
+        return not allowed_users[name]
+    end):each(function(name)
+        box.schema.user.drop(name)
+    end)
+    local allowed_roles = {
+        public = true,
+        replication = true,
+        super = true,
+    }
+
+    box.space._user:pairs():filter(function(tuple)
+        return tuple[4] == 'role'
+    end):map(function(tuple)
+        return tuple[3]
+    end):filter(function(name)
+        return not allowed_roles[name]
+    end):each(function(name)
+        box.schema.role.drop(name)
+    end)
+
+    os.execute(string.format('rm -rf *.snap *.xlog'))
+end
+
```

Comment:
I think we should clean _func and _cluster too.

Reply:
I guess adding asserts, will be enough:
```
assert(#box.space._func:select{} == 1)
assert(#box.space._cluster:select{} == 1)
```

Reply:
Will it fail the test in case on failing assert?

Reply:
Yes, the test fails.